experimental/sql/dialects/ibis_dialect.py

```diff
@@ -0,0 +1,257 @@
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from dataclasses import dataclass
+from typing import Any
+from xdsl.ir import Block, Region, Operation, SSAValue, ParametrizedAttribute, Data, MLContext, Attribute
+from xdsl.dialects.builtin import StringAttr, ArrayAttr, ArrayOfConstraint, IntegerAttr
+from xdsl.irdl import AttributeDef, OperandDef, ResultDef, RegionDef, SingleBlockRegionDef, irdl_attr_definition, irdl_op_definition, ParameterDef, AnyAttr, VarOperandDef, builder
+
+# This file defines a dialect that mirrors the internal data
+# structures of the ibis framework (https://ibis-project.org/docs/dev/).
+# In ibis, there exists an additional layer of abstractions called
+# `expressions`. Each operation in a query references the expressions of its
+# subqueries and is itself reference by an expression. This is a
+# remainder of the ibis UI that allows printing/executing/... of
+# expressions. Notice that the additional layer of indirection can make
+# ibis queries DAGs rather than trees. This only happens if the user
+# enters the query in a specific way, so this is also only a left-over
+# of the API. This dialect represents a cleaned up version of queries
+# without the expression layer for each node. In this version, all
+# queries are trees, so every part of the query is modeled as an
+# operation with regions for its respective subqueries. Hence, every
+# operation is a single region terminator and there are no SSAValues
```

Comment:
The fact that every operation is a single region terminator might not hold throughout the entire lifetime of this dialect. Imagine there is a selection with multiple predicates (T.a = 5 and T.b = 2). We could model this by having two equals operations in the predicate region of the selection. This might be a problem for later on tough.

Reply:
yes, I would put your wording as a disclaimer, everything is subject to evolution, we just want to avoid premature complexity.